### DISCUSSION TITLE
Live Stream

### DISCUSSION THREAD

**Discussion by dalogue1** [monkeymademe/CamUI#58] [General]
I really like the picamera2-WebUI application.  My end goal is to set up a live-stream of my birdfeeder on YouTube but this requires an RTMP or HLS stream. Is this very far from being able to be done using some parts of your application?

I can see my birdfeeder live on the picamera2-WebUI and somehow feel like it is not too big of a jump to publish an RTMP or HLS stream.  Maybe just a different python script.

What do you think?  

Again I really love your application.  I have been trying so hard to figure out how to use the camera and this made it so easy.

I have a 4GB RPi 5 with 1TB NVMe SSD and a Pi Camera V3 and your application runs great.

**Comment by monkeymademe** [comment]
Actually there is a way using the current system to stream. Use OBS and the feed URL http://robin.local:8080/video_feed then OBS does all the streaming lifting. If this was a pi4 I would say your out of luck I have tried to OBS there before and it was not good but maybe the pi5 is better? Worth a look?

Otherwise YT streaming is on my list but its low down there is a good few things I want to get working before I fuss with that (if its possible / worth doing quality wise)

**Reply by monkeymademe** [reply]
Ok future me here.... after the quickest of google searches I see it might... I stress 'might'.... be a easy setup but there might be consequences. Maybe the camera feed to the system will need disabled to stream to YT or the quality wont be spectacular. 

At the moment I am hoping I will manage to get Beta released this weekend after I was thinking of working on a playground for testing new features.  I think if its as easy as it looks, YT will be there. So in a few weeks (not long as a said in my last comment)

**Reply by dalogue1** [reply]
Thank you so much for your replies.  If that was possible to stream to YT it would absolutely be the cleanest means of creating a webcam that is used for purposes like mine.  Certainly I would be pleased to test anything.

Would the many adjustments to the camera settings still be present in the streaming version?   I would like that very much since that would allow someone to adjust the settings to what they want. (Your application makes this quite simple.)  They could then run a script to disable the camera feed to the system and enable the YT feed.  If they wanted to adjust camera settings they would just run a script to disable the YT feed and reenable the camera feed to the system.  (Later maybe these could be in a menu choice in the system but I don't want to get greedy.  Sorry.)

_I am very active in the BirdNET-Pi, group and am sure that others in the group would be interested in having a dual purpose for their Raspberry Pis.  Just so you know the BirdNET-Pi application is being run by many people around the world.  Look at the light green markers on the map of the Cornell University _BirdNET_ website https://app.birdweather.com/.   The active GitHub thread for BirdNET-PI is now https://github.com/Nachtzuster/BirdNET-Pi.  It has forked from the original founder of the project and is still very actively developed.  In case you do not know what BirdNET-Pi is here is the description from the BirdWeather site:  https://www.birdweather.com/birdnetpi. Their GitHub link still goes to the old archive but is an accurate description._

**Reply by dalogue1** [reply]
What would be even better is if your new version that is going to support two cameras on a RPi 5 then it might be possible to run a **regular** as well as a **NoIR** camera and switch between the two Raspberry Pi Camera 3 cameras depending upon time of day.  That way the Webcam would continue to show the video feed even at night.

**Reply by monkeymademe** [reply]
I just made the release for the new version v1.0.0 which will add pi5 dual camera support so you can have 2 cameras at the same time. Enjoy!

As for the YT... How this works I can't say for sure. If I get the stream working some camera settings can be made (brightness lens focus and the likes) can be made without restarting the camera feed. Rotation resolution (both capture resolution and sensor mode) needs a restart of the feed but I would assume that 'should' just recover. The Big question is can I have a feed to the system and YT at the same time?... 

I'll keep you informed

**Reply by dalogue1** [reply]
Thanks again.

I submitted a Bug Report on version 1.0.0

I noticed that you removed the ability to crop the image with a size and x or y offsets.  I saw your comment "Scalercrop is broken this should be hidden for now for the release and until its fixed".  I agree that it is better to remove it until it works.

Also, did the name of the image change from video_feed?  http://192.168.0.39:8081/video_feed no longer shows the image.  I think it is http://192.168.0.39:8081/video_feed/0 from looking at the code.

Sorry to sound critical since I really like the application.  Just testing what does or does not work.

**Reply by monkeymademe** [reply]
No worries... 

Yeah scalercrop needs another look... There was some confusion (on my part) over what the rendered resolution was and the sensor mode resolution was... I sorted that out now but it rendered the scalercrop feature half done.. It worked when everything was maxed out but when you fuss with the sensor mode it started to do strange things. Needs more work

You are right about the URL change for the video feed (admittedly I should have stated that in the release notes) but its one that has to be like that... The URL is like this now http://192.168.68.168:8080/video_feed_0 ... the 0 says which camera it is... on a pi5 you can have 0 and 1 because you can have 2 cameras connected (and running) at the same time. So even if you are on an older pi with only one connection its still port 0.

If you connect a USB camera which for now is not supported (I tried there is issues and limited support from picamera2) if it worked it would be _1 on an older pi or pi5 with one camera connected... or _2 if the pi5 was maxed out

**Reply by monkeymademe** [reply]
if you find the feed and right click and open in a new tab... that new tab will have the new video URL....

**Reply by dalogue1** [reply]
It was the URL you said!   i.e. video_feed_0

**Comment by monkeymademe** [comment]
I tried to look at streaming to youtube from picamera2 (outside of the webui) and I can't seem to get that to work. Something with the color format. Might be a while before I can implement this feature. Unless some example code out there has it working. Google says yes to picamera... but not picamera2 (which is a different beast)

**Reply by dalogue1** [reply]
Thanks for looking at it.
